libs/features/mas/src/checkout.js

```diff
@@ -49,13 +49,20 @@ export function Checkout({ settings, providers }) {
             entitlement,
             upgrade,
             modal,
+            fallbackStep,
             perpetual,
             promotionCode = defaultPromotionCode,
             wcsOsi,
             extraOptions,
             ...rest
         } = Object.assign(options, placeholder?.dataset ?? {}, overrides ?? {});  
+
+        const masFF3in1 = document.querySelector('meta[name=mas-ff-3in1]');
+        const is3in1 = Object.values(MODAL_TYPE_3_IN_1).includes(modal) && (!masFF3in1 || masFF3in1.content !== 'off');
```

Comment:
likely this whole logic could be moved to merch.js as we have access to whole information there.